### PR TITLE
Optimistic proposal creation logic

### PR DESCRIPTION
**Pull request by brickpop**
Changes:
- Adding `taiko-mono` as a submodule. Although we may safely remove it until OSx uses `^0.8.17`
- Remove startDate and endDate from all plugins
  - We are using `startDate = now` and `endDate = now + minDuration`
  - Multisig proposal expiration is kept
- Better interfacing with the TaikoL1 contract
  - Check wether the L2 is down
  - Check whether the last L2 synced block is old
- Adding L2 inactivity period and L2 aggregation grace period as part of the veto settings
- Removing unneeded `minProposerVotingPower`
- Keeping track of `minVetoRatio` per-proposal
- Using the L1-only census when the L2 is down
  - Storing wether the L2 was down when the proposal was created
  - Subtracting the bridged voting power when the L2 is unavailable or no L2 vetoes have been received, at all
- Registering the TaikoL1 contract address and the Bridge address
- Adding `bridgedVotingPower()`, `effectiveVotingPower()` and `isL2Available()` getters
- Extending the period before execution to allow for L2 vetoes to be bridged (when applicable)
- Read past token balances by timestamp not by block
- Allow emergency multisig proposals (duration = 0) to pass without L2 waiting
- Making the permission condition use the new ABI of `createProposal()`
- Using `^0.8.17` or newer everywhere
- Removing the redundant `getVotingToken()` getter
- Adding `struct InstallationParameters` to avoid stack too deep errors during installation
- Adding deployment env vars for the plugin settings

NOTE: Parsing the proposalID is part of PR #14, which should come before this one
NOTE 2: Tests are still WIP

### CODE REVIEW

**Review by carlosgj94 — APPROVED** [review]
Added some comment, but they are just questions I have that don't prevent the PR from being merged.

File: src/Multisig.sol [hunk 1]
```diff
@@ -43,24 +43,20 @@ contract Multisig is IMultisig, IMembership, PluginUUPSUpgradeable, ProposalUpgr
...
     /// @notice A container for the plugin settings.
     /// @param onlyListed Whether only listed addresses can create a proposal or not.
     /// @param minApprovals The minimal number of approvals required for a proposal to pass.
-    /// @param destinationMinDuration The minimum duration that the destination plugin will enforce.
+    /// @param destinationProposalDuration The minimum duration that the destination plugin will enforce.
     struct MultisigSettings {
         bool onlyListed;
         uint16 minApprovals;
-        uint64 destinationMinDuration;
+        uint64 destinationProposalDuration;
```
[the remark this answers is not in the document]
Comment (brickpop) [reply src/Multisig.sol:59]: @carlosgj94 That's why we have the permission condition, enforcing a minimum duration for the permission that the standard multisig has. This setting is meant to predefine de default "requested" duration without needing to pass a parameter each time

File: src/OptimisticTokenVotingPlugin.sol [hunk 2]
```diff
@@ -33,11 +35,13 @@ contract OptimisticTokenVotingPlugin is
...
     /// @param minDuration The minimum duration of the proposal vote in seconds.
-    /// @param minProposerVotingPower The minimum vetoing power required to create a proposal.
+    /// @param l2InactivityPeriod The age in seconds of the latest block, after which the L2 is considered unavailable.
+    /// @param l2AggregationGracePeriod The amount of extra seconds to allow for L2 veto bridging after `vetoEndDate` is reached.
     struct OptimisticGovernanceSettings {
         uint32 minVetoRatio;
         uint64 minDuration;
-        uint256 minProposerVotingPower;
+        uint64 l2InactivityPeriod;
+        uint64 l2AggregationGracePeriod;
```
[the remark this answers is not in the document]
Comment (brickpop) [reply src/OptimisticTokenVotingPlugin.sol:44]: I have drafted test values like 10 minutes (inactivity period) and 2 days (aggregation period), but these are env values that will need to be defined on the deployment script

**Comment by brickpop** [comment]
Merging the code changes, tests will be dealt with in a dedicated PR

**Pull request merged**